tests/rptest/services/redpanda.py

```diff
@@ -529,6 +545,16 @@ def cloud_storage_bucket(self):
         elif self.cloud_storage_type == CloudStorageType.ABS:
             return self._cloud_storage_azure_container
 
+    def gcp_iam_token(self):
+        try:
+            res = requests.get(
+                "http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/token",
+                headers={"Metadata-Flavor": "Google"})
```

Comment:
Does this handle retries implicitly? I think we might run into cases where the call fails, so we might need to make sure the call is retried.

Reply:
I also think we should `res.raise_for_status()` here, so if there is a non-2xx response we can log that.

Reply:
fixed both